packages/runtime/src/ctx.ts

```diff
@@ -181,32 +171,34 @@ export class Ctx implements Context {
    */
   optIntoASA (assetIndex: number, address: AccountAddress, flags: TxParams): void {
     const assetDef = this.getAssetDef(assetIndex);
-    const creatorAddr = assetDef.creator;
     makeAssetTransferTxnWithSuggestedParams(
       address, address, undefined, undefined, 0, undefined, assetIndex,
       mockSuggestedParams(flags, this.runtime.getRound()));
 
     const assetHolding: AssetHoldingM = {
-      amount: address === creatorAddr ? BigInt(assetDef.total) : 0n, // for creator opt-in amount is total assets
+      amount: 0n,
       'asset-id': assetIndex,
-      creator: creatorAddr,
-      'is-frozen': address === creatorAddr ? false : assetDef.defaultFrozen
+      creator: assetDef.creator,
+      'is-frozen': assetDef.defaultFrozen
```

Comment:
i think this was discussed before, assets are not frozen for creator. right ?
Please add a link here about this assumption

Reply:
The concept of Opting In for creator does not make sense, because the create created the asset, and therefore it cannot opt-out of the ASA as well. 
While creating an asset,  asset holding is added in creator's account will all assets & default frozen false. Means that this code (`optInToASA`) will never be executed for asset creator.

I will add a link about ASA's in `account.addApp` where we create asset holding for creator